src/printer-postcss.js

```diff
@@ -69,13 +69,19 @@ function genericPrint(path, options, print) {
         n.value.group.type === "value-value" &&
         n.value.group.group.type === "value-func" &&
         n.value.group.group.value === "extend";
+      const isComposed =
+        n.value.type === "value-root" &&
+        n.value.group.type === "value-value" &&
+        n.prop === "composes";
 
       return concat([
         n.raws.before.replace(/[\s;]/g, ""),
         n.prop,
         ":",
         isValueExtend ? "" : " ",
-        path.call(print, "value"),
+        isComposed
+          ? n.value.group.group.groups.join("")
```

Comment:
You want to do

```js
concat(path.map(print, "value", "group", "groups"))
```

Otherwise it doesn't go through the normal print mechanism.

It's not ideal that we have to reach into one level deep but without being on a computer I can't really troubleshot.

@azz, if you have a better idea, please let me know, otherwise i'll merge with my suggestion.